### PR TITLE
Add lacp no traffic tests

### PR DESCRIPTION
**Pull request by StepanVovkPLV**
Signed-off-by: Stepan Vovk <stepan.vovk@plvision.eu>

### CODE REVIEW

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/lacp/test_lacp_no_traffic.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
+import pytest
+
+from dent_os_testbed.lib.ip.ip_link import IpLink
+
```
Comment (yuriilisovskyi) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/lacp/test_lacp_no_traffic.py]: No need for newline

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/lacp/test_lacp_no_traffic.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    out = await IpLink.add(input_data=[{device: [{'device': bond,
+                                                  'type': 'bond',
+                                                  'mode': '802.3ad'}]}])
+    assert out[0][device]['rc'] == 0, 'Fail to create bond'
+
+    # 5. Add 9 ports to bond
+    out = await IpLink.set(input_data=[{device: [
+        {'device': f'swp{idx+1}',
+         'operstate': 'down'} for idx in range(max_ports_in_lag + 1)]}])
+    assert out[0][device]['rc'] == 0, 'Fail to set port to up state'
```
Comment (yuriilisovskyi) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/lacp/test_lacp_no_traffic.py]: to down state*

**Review by yuriilisovskyi — APPROVED** [review]

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/lacp/test_lacp_no_traffic.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    max_bonds = 64
+    # 1. Create max 'number of  bond devices
+    cmd = [{'device': f'bond_{idx}',
+            'type': 'bond',
+            'mode': '802.3ad'} for idx in range(max_bonds)]
+    out = await IpLink.add(input_data=[{device: cmd}])
+    # 2. Verify it is possible to create bond devices
+    assert out[0][device]['rc'] == 0, 'Fail to create bond(s)'
+
+    _, num_of_ports = await dent_devices[0].run_cmd('ifconfig  -a | grep -Eo "swp[0-9]+" | wc -l')
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/lacp/test_lacp_no_traffic.py]: please also check `rc`

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/lacp/test_lacp_no_traffic.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+    # 1. Create bond device
+    out = await IpLink.add(input_data=[{device: [{'device': bond,
+                                                  'type': 'bond',
+                                                  'mode': '802.3ad'}]}])
+    assert out[0][device]['rc'] == 0, 'Fail to create bond'
+
+    # 5. Add 9 ports to bond
+    out = await IpLink.set(input_data=[{device: [
+        {'device': f'swp{idx+1}',
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/lacp/test_lacp_no_traffic.py]: I would suggest getting the port list via `IpLink.show()`, and not using an `'swp'` string. The port on DUT might have a different naming convention (`sw1p1`).

**Review by SerhiyBoikoPLV — CHANGES_REQUESTED** [review]

**Review by SerhiyBoikoPLV — CHANGES_REQUESTED** [review]
please add `__init__.py` to `./functional/lacp/` folder, because otherwise pytest does not see these LACP tests

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/lacp/test_lacp_no_traffic.py [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    assert int(num_of_ports) >= 1, 'Fail getting port on devices'
+
+    out = await IpLink.set(input_data=[{device: [
+        {'device': port,
+         'operstate': 'down'} for port in device_ports]}])
+    assert out[0][device]['rc'] == 0, 'Fail to set port(s) to down state'
+    # 2. Enslave all ports to different bonds
+    out = await IpLink.set(input_data=[{device: [
+        {'device': port[1],
+         'master': f'bond_{port[0]}'} for port in enumerate(device_ports)]}])
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/lacp/test_lacp_no_traffic.py]:
```suggestion
         'master': f'bond_{idx}'} for idx, port in enumerate(device_ports)]}])
```

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/lacp/test_lacp_no_traffic.py [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+    out = await IpLink.set(input_data=[{device: [
+        {'device': port,
+         'master': bond} for port in device_ports[:max_ports_in_lag + 1]]}])
+    assert 'RTNETLINK answers: No space left on device' in out[0][device]['result'], 'Can enslave more ports to bond'
+
+    out = await IpLink.set(input_data=[{device: [
+        {'device': port,
+         'operstate': 'up'} for port in device_ports[:max_ports_in_lag + 1]]}])
+    assert out[0][device]['rc'] == 0, f'Fail enslaving {max_ports_in_lag} ports to bond'
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/lacp/test_lacp_no_traffic.py]: please change assert message, you are not enslaving any ports here

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/lacp/test_lacp_no_traffic.py [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+    # 5. Add 9 ports to bond
+    out = await IpLink.set(input_data=[{device: [
+        {'device': port,
+         'operstate': 'down'} for port in device_ports[:max_ports_in_lag + 1]]}])
+    assert out[0][device]['rc'] == 0, 'Fail to set port to down state'
+
+    out = await IpLink.set(input_data=[{device: [
+        {'device': port,
+         'master': bond} for port in device_ports[:max_ports_in_lag + 1]]}])
+    assert 'RTNETLINK answers: No space left on device' in out[0][device]['result'], 'Can enslave more ports to bond'
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/lacp/test_lacp_no_traffic.py]:
```suggestion
    assert 'RTNETLINK answers: No space left on device' in out[0][device]['result'], \
        'Failed to enslave ports to bond'
```

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/lacp/test_lacp_no_traffic.py [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+        {'device': port[1],
+         'master': f'bond_{port[0]}'} for port in enumerate(device_ports)]}])
+
+    # 3. Verify it is possible to enslave all ports to bond(s)
+    assert out[0][device]['rc'] == 0, 'Fail to enslave port(s) to bond'
+
+    out = await IpLink.set(input_data=[{device: [
+        {'device': port,
+         'operstate': 'up'} for port in device_ports]}])
+    assert out[0][device]['rc'] == 0, 'Fail to set bond(s) to up state'
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/lacp/test_lacp_no_traffic.py]:
```suggestion
    assert out[0][device]['rc'] == 0, 'Fail to set ports to up state'
```

**Review by SerhiyBoikoPLV — CHANGES_REQUESTED** [review]

**Review by SerhiyBoikoPLV — APPROVED** [review]

**Pull request merged**